nvme.c

```diff
@@ -5373,6 +5454,676 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
 	return err;
 }
 
+int get_reg_size(int offset)
+{
+	return nvme_is_64bit_reg(offset) ? sizeof(uint64_t) : sizeof(uint32_t);
+}
+
+static inline bool is_set_reg(struct get_reg_config *cfg, int offset)
+{
+	switch (offset) {
+	case NVME_REG_CAP:
+		if (cfg->cap)
+			return true;
+		break;
+	case NVME_REG_VS:
+		if (cfg->vs)
+			return true;
+		break;
+	case NVME_REG_INTMS:
+		if (cfg->intms)
+			return true;
+		break;
+	case NVME_REG_INTMC:
+		if (cfg->intmc)
+			return true;
+		break;
+	case NVME_REG_CC:
+		if (cfg->cc)
+			return true;
+		break;
+	case NVME_REG_CSTS:
+		if (cfg->csts)
+			return true;
+		break;
+	case NVME_REG_NSSR:
+		if (cfg->nssr)
+			return true;
+		break;
+	case NVME_REG_AQA:
+		if (cfg->aqa)
+			return true;
+		break;
+	case NVME_REG_ASQ:
+		if (cfg->asq)
+			return true;
+		break;
+	case NVME_REG_ACQ:
+		if (cfg->acq)
+			return true;
+		break;
+	case NVME_REG_CMBLOC:
+		if (cfg->cmbloc)
+			return true;
+		break;
+	case NVME_REG_CMBSZ:
+		if (cfg->cmbsz)
+			return true;
+		break;
+	case NVME_REG_BPINFO:
+		if (cfg->bpinfo)
+			return true;
+		break;
+	case NVME_REG_BPRSEL:
+		if (cfg->bprsel)
+			return true;
+		break;
+	case NVME_REG_BPMBL:
+		if (cfg->bpmbl)
+			return true;
+		break;
+	case NVME_REG_CMBMSC:
+		if (cfg->cmbmsc)
+			return true;
+		break;
+	case NVME_REG_CMBSTS:
+		if (cfg->cmbsts)
+			return true;
+		break;
+	case NVME_REG_CMBEBS:
+		if (cfg->cmbebs)
+			return true;
+		break;
+	case NVME_REG_CMBSWTP:
+		if (cfg->cmbswtp)
+			return true;
+		break;
+	case NVME_REG_NSSD:
+		if (cfg->nssd)
+			return true;
+		break;
+	case NVME_REG_CRTO:
+		if (cfg->crto)
+			return true;
+		break;
+	case NVME_REG_PMRCAP:
+		if (cfg->pmrcap)
+			return true;
+		break;
+	case NVME_REG_PMRCTL:
+		if (cfg->pmrctl)
+			return true;
+		break;
+	case NVME_REG_PMRSTS:
+		if (cfg->pmrsts)
+			return true;
+		break;
+	case NVME_REG_PMREBS:
+		if (cfg->pmrebs)
+			return true;
+		break;
+	case NVME_REG_PMRSWTP:
+		if (cfg->pmrswtp)
+			return true;
+		break;
+	case NVME_REG_PMRMSCL:
+		if (cfg->pmrmscl)
+			return true;
+		break;
+	case NVME_REG_PMRMSCU:
+		if (cfg->pmrmscu)
+			return true;
+		break;
+	default:
+		break;
+	}
+
+	return false;
+}
+
+static int get_register_properties(int fd, void **pbar, struct get_reg_config *cfg)
```

Comment:
Seems like there is significant duplication with `nvme_get_properties()` (`nvme show-regs`), as well as `nvme get-property`. Is there any way we could combine them together? Maybe `nvme show-regs` and `nvme get-property` synthesize a `struct get_reg_config`, and `nvme get-property` forces `fabrics = true`?
Similar for `nvme set-property` and `nvme set-reg`.